src/ui/src/custom_components/core.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Declare custom components
+ *
+ * @param customComponents list of components in key-value form
+ */
+export function declareCustomComponents(
+	customComponents: Record<string, unknown>,
+) {
+	Object.entries(customComponents).forEach(([key, customComponent]) => {
+		_auditCustomComponents(key, customComponent);
+	});
+
+	return customComponents;
+}
+
+/**
+ * Audit the validity of custom components
+ *
+ * @param key
+ * @param customComponent
+ */
+export function _auditCustomComponents(key: string, customComponent: unknown) {
+	// Checks that the key contains only alphanumeric characters and underscores without capital letters
+	// @see https://github.com/writer/writer-framework/issues/517
+	const regex = /^[a-z0-9_]+$/;
+	const is_valid_key = regex.test(key);
+	if (!is_valid_key) {
+		throw Error(
+			`custom component key '${key}' is not valid. custom component should be declared with only alphanumeric lowercase and _.`,
+		);
```

Comment:
Question, do we really want to stop the application here ? Do you think `console.warn` could be enough ?